### PR TITLE
Add hello world rstest and cucumber coverage

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- add rstest fixtures covering hello world CLI parsing, validation, and command helpers
- extend the cucumber world with temp-dir config helpers and an environment precedence scenario
- document the testing approach and update the example checklist

## Testing
- make fmt
- make lint
- make test

------
https://chatgpt.com/codex/tasks/task_e_68d7c8ad90e8832294eb8ba9295d70f4

## Summary by Sourcery

Add comprehensive rstest unit tests and enhance cucumber-based behavioural tests for the `hello_world` example, introduce working-directory and env/config-file helpers, update documentation, and bump test-related dependencies.

New Features:
- Add parameterized rstest fixtures for CLI parsing, validation, delivery-mode logic, and edge-cases in the `hello_world` example
- Introduce Cucumber-world helpers for temporary working directories, configuration file writing, and environment variable injection
- Add a Cucumber scenario to verify configuration file < environment variable < CLI argument precedence

Enhancements:
- Replace legacy build_plan and build_take_leave_plan tests with structured rstest cases covering validation errors, delivery modes, and trimming logic

Build:
- Add `tempfile`, `cap-std`, and `gherkin` dependencies to the example’s Cargo.toml for test support

Documentation:
- Document the combined rstest and cucumber testing approach in design.md and user-guide, and update the example README checklist

Tests:
- Extend Cucumber steps and feature files to leverage new helpers and cover environment/config-file scenarios

### CODE REVIEW

File: examples/hello_world/tests/steps/global.rs [hunk 1]
```diff
@@ -55,3 +55,21 @@ pub fn stdout_contains(world: &mut World, expected: String) {
...
+#[given(expr = "the environment contains {string} = {string}")]
+#[expect(
+    clippy::needless_pass_by_value,
+    reason = "Cucumber step signature requires owned String"
+)]
+pub fn environment_contains(world: &mut World, key: String, value: String) {
+    world.set_env(key, value);
+}
+
+#[given("the hello world config file sets file defaults")]
```
[the remark this answers is not in the document]
Comment (leynos) [reply examples/hello_world/tests/steps/global.rs]: Has this now been resolved?

File: examples/hello_world/tests/steps/global.rs [hunk 2]
```diff
@@ -55,3 +55,21 @@ pub fn stdout_contains(world: &mut World, expected: String) {
 pub fn stderr_contains(world: &mut World, expected: String) {
     world.assert_stderr_contains(&expected);
 }
+
+#[given(expr = "the environment contains {string} = {string}")]
```
[the remark this answers is not in the document]
Comment (leynos) [reply examples/hello_world/tests/steps/global.rs:60]: Has this now been resolved?

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

- Factor out the repeated CommandLine::try_parse_from invocations and assertion matching into an rstest fixture or helper to DRY up the CLI parsing tests.
- Merge the two blank-value validation tests for `GreetCommand` (preamble and punctuation) into a single parameterized rstest so you can centralize the error assertions and reduce boilerplate.
- Consider abstracting the Cucumber world’s env-setting and config-writing logic into smaller helper methods or a trait to keep the `run_example` path focused solely on execution and output assertions.
